cli/src/pcluster/validators/directory_service_validators.py

```diff
@@ -73,13 +75,28 @@ def _validate(self, domain_name):
 class PasswordSecretArnValidator(Validator):
     """PasswordSecretArn validator."""
 
-    def _validate(self, password_secret_arn):
-        """Validate that PasswordSecretArn contains an ARN of a readable secret in AWS Secrets Manager."""
+    def _validate(self, password_secret_arn: str, region: str):
+        """Validate that PasswordSecretArn contains a valid ARN for the given region.
+
+        In particular, the ARN should be one of the following resources:
+         1. a readable secret in AWS Secrets Manager, which is supported in all regions but us-isob-east-1.
+         2. a readable parameter in SSM Parameter Store, which is supported only in us-isob-east-1.
+        """
         try:
             # We only require the secret to exist; we do not validate its content.
-            AWSApi.instance().secretsmanager.describe_secret(password_secret_arn)
+            arn_components = Arn.parse(password_secret_arn)
+            service, resource = arn_components.service, arn_components.resource
+            if service == "secretsmanager" and resource == "secret" and region != "us-isob-east-1":
```

Comment:
It seems that this block could provide more specific information for certain situations.  If the user is trying to use secretsmanager in the isob region, we could produce an error to inform them of that instead of a generic unsupported message.

Reply:
May you please provide a specific example? What additional infornmation are you thinking about?

Reply:
You mean: explaining the customer that SecretsManager is not supported in us-isob-east-1?
In this case, I agree with you we may be more explicit here.
We will address this in a follow up activity.